Add kumactl to arkade get

Shikachuu opened the pull request:
## Description
Related to #571, it adds the Kuma cli tool to `arkade get`

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
- [X] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md)) #571


## How Has This Been Tested?
After build I ran `arkade get kumactl` and the install worked on the following systems:
- Debian
- CentOS
- Ubuntu
- RedHat

**It hasn't been tested on darwin, since currently I do not own a Mac.**

## Are you a GitHub Sponsor (Yes/No?)

<!--- Check at https://github.com/sponsors/alexellis         -->
<!--- Sponsors get priority because they support the project -->

- [ ] Yes
- [X] No

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [X] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [X] All new and existing tests passed.

<!--- In case it is a new application -->
- [x] I have tested this on arm, or have added code to prevent deployment

alexellis commented on pkg/get/tools.go:
```diff
@@ -1823,5 +1823,23 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Name}}
+			Name:        "kumactl",
+			Version:     "1.3.1",
+			Description: "kumactl is a CLI to interact with Kuma and its data",
+			URLTemplate: `
```
Looks good. We will need a unit test adding though. I don't mind if you do that as commit 2 or reset into one commit.

Shikachuu replied:
I added some test for darwin x64 and linux x64 (only these arch + os combos are supported).

alexellis reviewed (changes requested):
Please see my message.

You can test these by running:

```
# Override each variation, including one which is unavailable:
arkade get --os ... --arch ...

# Print the architecture:
file ./kumactl
```

alexellis commented on pkg/get/tools.go:
```diff
@@ -1823,5 +1823,23 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Name}}
+			{{- else if eq .OS "darwin" -}}
+			{{$osStr = "darwin"}}
+			{{- end -}}
+			https://download.konghq.com/mesh-alpine/{{.Repo}}-{{.Version}}-{{$osStr}}-amd64.tar.gz`,
```
So, if you use armv7l as the `--arch` override, you will get the wrong binary.

We need to template the `amd64` portion of this string. Why? So that's empty unless it's set otherwise.

For invalid architectures and operating systems we should cause a 404 in the URL.

Also, do they really not have a Windows binary for their CLI?

Shikachuu replied:
They do not have a windows binary. :disappointed: 
I'll fix the arch problems asap, thanks for the note! :smile:

alexellis commented:
The video I have on rebasing commits into one might be useful. Derek shared a link for it, you can squash the no dco error at the same time.

Shikachuu commented:
> The video I have on rebasing commits into one might be useful. Derek shared a link for it, you can squash the no dco error at the same time.

It was a little more complex since I auto-merged master before, so had to do some tinkering but it came up just fine. :smile:

alexellis commented:
Please rebase @Shikachuu

alexellis commented:
We need a quick rebase on this one @Shikachuu - please also see if there's anything else you want to do for consistency, or if it's good to go.

alexellis reviewed (approved):
Approved

alexellis commented:
@Shikachuu please can you rebase due to other merged PRs conflicting with these changes?

Shikachuu commented:
Done! :heavy_check_mark: 
But the e2e url check failed...yet the URL is alive and well.

alexellis commented:
>But the e2e url check failed...yet the URL is alive and well.

Can you work out why?

alexellis commented:
Run the command locally, and you'll see that it's forbidden:

```
curl -i --head -L https://download.konghq.com/mesh-alpine/kuma-1.4.1-ubuntu-amd64.tar.gz
```

This is probably because Kuma is using an S3 bucket, perhaps it disallows HEAD methods?

Shikachuu commented:
> Run the command locally, and you'll see that it's forbidden:
> 
> ```
> curl -i --head -L https://download.konghq.com/mesh-alpine/kuma-1.4.1-ubuntu-amd64.tar.gz
> ```
> 
> This is probably because Kuma is using an S3 bucket, perhaps it disallows HEAD methods?
> 
> https://stackoverflow.com/questions/27469801/403-forbidden-error-on-s3-rest-api-head-request

As we talked about this, the S3 bucket disallows HEAD requests, so I added a skip mechanism to the e2e tests, and added `kumactl` to it. [Also I created an issue in the kuma main repo as well.](https://github.com/kumahq/kuma/issues/4019)

dannysauer commented:
I'm linking [my comment with workarounds](https://github.com/kumahq/kuma/issues/4019#issuecomment-1075399577) in a new comment here since I forgot to tag this issue for cross-thread awareness. :)  If there's any feedback on feasibility of those workarounds which might help other people, I guess the Kong thread might be the more discoverable place to provide such feedback.  Thanks!

alexellis reviewed (approved):
Approved
